pkg/codegen/dotnet/gen.go

```diff
@@ -265,11 +265,16 @@ func (mod *modContext) typeString(t schema.Type, qualifier string, input, state,
 			typ += "Result"
 		}
 	case *schema.ResourceType:
-		typ = mod.tokenToNamespace(t.Token, "")
-		if typ != "" {
-			typ += "."
+		if strings.HasPrefix(t.Token, "pulumi:providers:") {
+			pkgName := strings.TrimPrefix(t.Token, "pulumi:providers:")
+			typ = fmt.Sprintf("Pulumi.%s.Provider", Title(pkgName))
```

Comment:
`Title` won't quite cut it for cases like `azure-nextgen`. You likely need to use `tokenToNamespace`.